corehq/apps/domain/forms.py

```diff
@@ -2377,3 +2383,91 @@ def __init__(self, defaults=None, disable_input=False, **kwargs):
                     css_class="disabled"
                 )
             )
+
+
+class ManageAppReleasesForm(forms.Form):
+    app_id = forms.ChoiceField(label=ugettext_lazy("Application"), choices=(), required=False)
+    location_id = forms.ChoiceField(label=ugettext_lazy("Location"), choices=(), required=False)
+    version = forms.IntegerField(label=ugettext_lazy('Version'), required=False)
+
+    def __init__(self, request, domain, *args, **kwargs):
+        self.domain = domain
+        super(ManageAppReleasesForm, self).__init__(*args, **kwargs)
+        self.fields['app_id'].choices = self.app_id_choices()
+        if request.GET.get('app_id'):
+            self.fields['app_id'].initial = request.GET.get('app_id')
+        self.fields['location_id'].choices = self.location_id_choices()
+        if request.GET.get('location_id'):
+            self.fields['location_id'].initial = request.GET.get('location_id')
+        if request.GET.get('version'):
+            self.fields['version'].initial = request.GET.get('version')
+        self.helper = FormHelper()
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.form_tag = False
+
+        self.helper.layout = crispy.Layout(
+            crispy.Field('app_id', id='app-id-search-select'),
+            crispy.Field('location_id', id='location-search-select', css_class="ko-select2"),
+            crispy.Field('version', id='version-input'),
```

Comment:
Can this use a select2 dropdown and show the version number and comment?

This commit sets up a version select2 that changes its options based on the value of an application dropdown: https://github.com/dimagi/commcare-hq/pull/23521/commits/47e80043d0ba461c6276fce3d8ad134d17ea2d88

Reply:
Thanks for suggesting that, i see that that js change now returns [build_id instead of version](https://github.com/dimagi/commcare-hq/pull/23521/files#diff-62b3073366d0a0237a9467da2df883b4R31). This is good. I can remove some logic here to find out build from version and also validation around it.
Though i think we broke the implementation in the PR you linked coz it still relies on getting a version number.

Reply:
while making the changes i realised it was better to just update the parent js to use version instead of build id as the key since we need to pass on version anyway so we would need both build id and version.. plus the other place where this was changed for `AppTranslationForm` also needs version and not build id.

Reply:
[commit](https://github.com/dimagi/commcare-hq/pull/23680/commits/f6bf3a72606fa4f664b72efa178f330c35d51829)